src/doc/rustc/src/platform-support/unknown-uefi.md

```diff
@@ -252,3 +251,146 @@ pub extern "C" fn main(_h: efi::Handle, st: *mut efi::SystemTable) -> efi::Statu
     efi::Status::SUCCESS
 }
 ```
+
+## Rust std for UEFI
+This section contains information on how to use std on UEFI. It will
+continuously evolve, at least untill everything is merged upstream.
+
+### Build std
+The building std part is pretty much the same as the official [docs](https://rustc-dev-guide.rust-lang.org/getting-started.html).
+The linker that should be used is `rust-lld`. Here is a sample `config.toml`:
+```toml
+[rust]
+lld = true
+
+[target.x86_64-unknown-uefi]
+linker = "rust-lld"
+```
+Then just build using `x.py`:
+```sh
+./x.py build --target x86_64-unknown-uefi
+```
+
+### Std Requirements
+The current std has a few basic requirements to function:
+1. The signature of entry function as specified in section 4.1 in UEFI
+   Specification. Also, the name of this function is `efi_main`.
+2. Memory Allocation Services (`EFI_BOOT_SERVICES.AllocatePool()` and
+   `EFI_BOOT_SERVICES.FreePool()`) are available.
```

Comment:
`efi_main` is picked by the target configuration as PE32+ entry-point. So this is all internal to `std`, I don't think this needs to be documented as public API. If you link this as application, the name stays fully internal.

Secondly, the memory allocation services are mandatory by the UEFI specification. I think this can be dropped as requirement, right?

Reply:
Well, they are mandatory, however, they will not be functional until Boot Services are read (I think), so it needs those services to be usable by the time a binary using std runs.